pytorch_lightning/trainer/trainer.py

```diff
@@ -760,31 +762,61 @@ def __attach_dataloaders(self, model, train_dataloader, val_dataloaders, test_da
 
     def init_optimizers(
             self,
-            optimizers: Union[Optimizer, Tuple[List, List], List[Optimizer], Tuple[Optimizer]]
-    ) -> Tuple[List, List]:
+            optim_conf: Union[Optimizer, Sequence[Optimizer], Dict, Sequence[Dict], Tuple[List, List]]
```

Comment:
just thinking if giving such freedom with a dictionary if we require certain field to be presented, right?
so `nametuple` could work too, but maybe `dict` is a good option... :]

Reply:
My idea is that we might find out later that we want extra keys to be implemented.
Like switching optimizers (multiple phases optimization), or allowing finetuning as part of one "fit" call...
Many ideas might be possible. A dictionary gives us this flexibility.
In "training_step" we return a dictionary with a required key "loss".